python/hail/api1/context.py

```diff
@@ -843,16 +843,30 @@ def import_vcf(self, path, force=False, force_bgz=False, header_file=None, min_p
     @handle_py4j
     @record_method
     @typecheck_method(path=oneof(strlike, listof(strlike)),
+                      key_expr=strlike,
+                      annotation_types=listof(Type),
+                      number_of_annotations=integral,
+                      annotation_headers=nullable(listof(strlike)),
                       min_partitions=nullable(integral),
                       drop_samples=bool,
                       cell_type=nullable(Type),
-                      missing=strlike,
-                      has_row_id_name=bool)
-    def import_matrix(self, path, min_partitions=None, drop_samples=False, cell_type=None, missing="NA", has_row_id_name = False):
+                      missing=strlike)
+    def import_matrix(self, path, key_expr, annotation_types, number_of_annotations=1, annotation_headers=None,
+                      min_partitions=None, drop_samples=False, cell_type=None, missing="NA"):
         """
         :param path: File(s) to read. Currently, takes 1 header line of column ids and subsequent lines of rowID, data... in TSV form where data can be parsed as an integer.
         :type path: str or list of str
 
+        :param str key_expr: Expression to use for the row key.
+
+        :param annotation_types: List of types to use for the annotation fields. Must be one of: TInt32, TInt64, TFlost32, TFlost64, TString.
+        :type min_partitions: list of Type
```

Comment:
Guessing you meant `annotation_types` here? (otherwise dup with `min_partitions` on line 868)